### DISCUSSION TITLE
How to register MR to CT of the same person?

### DISCUSSION THREAD

**Discussion by kurmukovai** [dipy/dipy#3023] [Registration]
I am trying to register **Flair image** to **CT image** (same person, different Orientation Matrix, and Patient Location). Basically the only thing that registration algorithm need to do is to shift and rotate a Flair image in geometrical space. And it fails to do so, well, I fail to make it work.

I've tried this tutorial https://dipy.org/documentation/1.1.1./examples_built/affine_registration_3d/#example-affine-registration-3d

I've also tried to use dipy.align.imaffine.transform_origins and 
 dipy.align.imaffine.transform_geometric_centers, since they kinda looks like one I need, both failed.

```python
static_data, static_affine = load_nifti(p_static)
static = static_data
static_grid2world = static_affine


moving_data, moving_affine = load_nifti(p_moving)
moving = moving_data
moving_grid2world = moving_affine

affinemap = transform_geometric_centers(static, static_grid2world, moving, moving_grid2world)
result = affinemap.transform(moving)
```
```
Affine static
[[   1.     0.     0.  -163.5]
 [   0.     1.     0.  -163.5]
 [   0.     0.     1.  -150. ]
 [   0.     0.     0.     1. ]]

Affine moving
[[   0.99997997   -0.00112       0.00634    -120.23751068]
 [   0.            0.98477       0.17386    -105.13677216]
 [  -0.00643816   -0.17385653    0.98475033  -47.00952148]
 [   0.            0.            0.            1.        ]]
```

PS This might be the wrong place to ask, it would be cool if anyone point me to the right place.

**Comment by arokem** [comment]
Thanks for getting in touch. Just to clarify: when you say it "fails" -- what exactly do you mean? Do you mean that you get a result, but it doesn't seem well-registered? Could you post some pictures of what that looks like?

**Comment by kurmukovai** [comment]
Sure, this is for example result of `transform_geometric_centers`

![Screenshot from 2020-03-02 23-01-09](https://user-images.githubusercontent.com/14237010/75751919-ae2da480-5d38-11ea-8ff8-04fc3980d029.png)

I have a question, it seems like both transform_geometric_centers and transform_origins do not use information about voxel's size. Since by default `load_nifti` does not return it  (`return_voxsize=False`), but even if I do read it from nifti, how am I suppose to force these functions to use it? Because my images have quite different voxel sizes.

PS I could upload these CT, MR nifties somewhere if it helps.

**Comment by pcamach2** [comment]
@kurmukovai if the difference in voxel size between the CT and FLAIR is causing issues, you can try reslicing before transforming

[https://dipy.org/documentation/1.2.0./examples_built/reslice_datasets/](https://dipy.org/documentation/1.2.0./examples_built/reslice_datasets/)
